src/app/reporting/LoggingManagement.cpp

```diff
@@ -0,0 +1,1306 @@
+ *   subsystem with provided resources.
+ *
+ * Initialize the LoggingManagement with an array of LogStorageResources.  The
+ * array must provide a resource for each valid priority level, the elements
+ * of the array must be in increasing numerical value of priority (and in
+ * decreasing priority); the first element in the array corresponds to the
+ * resources allocated for the most critical events, and the last element
+ * corresponds to the least important events.
+ *
+ * @param[in] inMgr         ExchangeManager to be used with this logging subsystem
+ *
+ * @param[in] inNumBuffers  Number of elements in inLogStorageResources array
+ *
+ * @param[in] inLogStorageResources  An array of LogStorageResources for each priority level.
+ *
+ * @note This function must be called prior to the logging being used.
+ */
+void LoggingManagement::CreateLoggingManagement(Messaging::ExchangeManager * apMgr, size_t aNumBuffers,
+                                                const LogStorageResources * const apLogStorageResources)
+{
+    new (&sInstance) LoggingManagement(apMgr, aNumBuffers, apLogStorageResources);
+}
+
+/**
+ * @brief Perform any actions we need to on shutdown.
+ */
+void LoggingManagement::DestroyLoggingManagement(void)
+{
+    Platform::CriticalSectionEnter();
+    sInstance.mState        = kLoggingManagementState_Shutdown;
+    sInstance.mpEventBuffer = NULL;
+    Platform::CriticalSectionExit();
+}
+
+/**
+ * @brief Set the ExchangeManager to be used with this logging subsystem.  On some
+ *   platforms, this may need to happen separately from CreateLoggingManagement() above.
+ *
+ * @param[in] inMgr         ExchangeManager to be used with this logging subsystem
+ */
+CHIP_ERROR LoggingManagement::SetExchangeManager(Messaging::ExchangeManager * apMgr)
+{
+    mpExchangeMgr = apMgr;
+    return CHIP_NO_ERROR;
+}
+
+/**
+ * @brief
+ *   LoggingManagement constructor
+ *
+ * Initialize the LoggingManagement with an array of LogStorageResources.  The
+ * array must provide a resource for each valid priority level, the elements
+ * of the array must be in increasing numerical value of priority (and in
+ * increasing priority); the first element in the array corresponds to the
+ * resources allocated for least important events, and the last element
+ * corresponds to the most critical events.
+ *
+ * @param[in] apMgr         ExchangeManager to be used with this logging subsystem
+ *
+ * @param[in] aNumBuffers  Number of elements in inLogStorageResources array
+ *
+ * @param[in] apLogStorageResources  An array of LogStorageResources for each priority level.
+ *
+ */
+
+LoggingManagement::LoggingManagement(Messaging::ExchangeManager * apMgr, size_t aNumBuffers,
+                                     const LogStorageResources * const apLogStorageResources)
+{
+    CircularEventBuffer * current = NULL;
+    CircularEventBuffer * prev    = NULL;
+    CircularEventBuffer * next    = NULL;
+    size_t i, j;
```

Comment:
Please reduce scope for these: j should be declared within the for loop and i inside.

`i` should have a better name that explains its usage - as it stands code is hard to follow.

Reply:
same with other variables like next,prev/current - all seem to only be used inside the for loop.

Reply:
Done, thanks